### PR TITLE
Thumbnail

### PR DESCRIPTION
**Pull request by mokaddem**
### :warning: WARNING :warning:
- This PR introduces a new dependency ``php-gd``
- All install files should have been updated accordingly
- Also related to https://github.com/MISP/misp-packer/pull/15

## Features

- Asynchronous pictures loading in ``event view``
- Improved Image modal view allowing large pictures
- Added ``viewPicture`` endpoint
- Integrated picture re-sampling allowing to resize picture on-demand
- Added support in diagnostic

### CODE REVIEW

**Comment by adulau** [comment]
This pull request **fixes 1 alert** when merging 749c9ea54416280d8e6c73faa802c8078b5a99c5 into 695771eb44ccb650b35bd8a973972b7c2996cd99 - [view on LGTM.com](https://lgtm.com/projects/g/MISP/MISP/rev/pr-332d4c6741260cec5f0d77c10728a3d9eeb581c3)

**fixed alerts:**

* 1 for Missing variable declaration

---

*Comment posted by [LGTM.com](https://lgtm.com)*

File: app/Controller/AttributesController.php [hunk 1]
```diff
@@ -1189,7 +1189,85 @@ public function view($id)
...
+        }
+        $conditions = array('conditions' => array('Attribute.id' => $id), 'withAttachments' => true, 'flatten' => true);
+        $conditions['includeAllTags'] = false;
+        $conditions['includeAttributeUuid'] = true;
+        $attribute = $this->Attribute->fetchAttributes($this->Auth->user(), $conditions);
+        if (empty($attribute)) {
+            throw new MethodNotAllowedException('Invalid attribute');
+        }
+        $attribute = $attribute[0];
+        if ('attachment' != $attribute['Attribute']['type'] && 'malware-sample' != $attribute['Attribute']['type'] ) {
```
Comment (iglocska) [inline app/Controller/AttributesController.php]: Why not include this in the conditions already? Also, malware sample should NOT be a balid type for thumbnails.

File: app/View/Elements/Events/View/value_field.ctp [hunk 2]
```diff
@@ -2,10 +2,16 @@
   $sigDisplay = $object['value'];
   if ('attachment' == $object['type'] || 'malware-sample' == $object['type'] ) {
     if ($object['type'] == 'attachment' && isset($object['image'])) {
-      $extension = explode('.', $object['value']);
-      $extension = end($extension);
-      $uri = 'data:image/' . strtolower(h($extension)) . ';base64,' . h($object['image']);
-      echo '<img class="screenshot screenshot-collapsed useCursorPointer" src="' . $uri . '" title="' . h($object['value']) . '" />';
+        if (extension_loaded('gd')) {
+            $img = '<it class="fa fa-spin fa-spinner" style="font-size: large; left: 50%; top: 50%;"></it>';
+            $img .= '<img class="screenshot screenshot-collapsed useCursorPointer img-rounded" src="' . $baseurl . '/attributes/viewPicture/' . h($object['id']) . '/1' . '" title="' . h($object['value']) . '" onload="$(this).parent().find(\'.fa-spinner\').remove();"/>';
```
Comment (iglocska) [inline app/View/Elements/Events/View/value_field.ctp]: This looks great, however, keep in mind that the image is still loaded, b64 encoded and passed to the view - it just isn't transferred to the user. Perhaps also change the bahaviour on the controller side (EventsController->view() / EventsController->__viewUI() / EventsController->viewEventAttributes() - or whatever it's called).

**Review by iglocska — CHANGES_REQUESTED** [review]

File: app/Controller/AttributesController.php (hunk 1 above)
Comment (mokaddem) [reply app/Controller/AttributesController.php]: Copy pasta master... :1st_place_medal: 
You are totally right!

File: app/View/Elements/Events/View/value_field.ctp (hunk 2 above)
Comment (mokaddem) [reply app/View/Elements/Events/View/value_field.ctp]: Sure! Especially that this PR aims to improve the perf...
will have a look.

**Comment by adulau** [comment]
This pull request **fixes 1 alert** when merging 2ac6d2434e951a33a630517a900d858772793b65 into 43f0010cab9776d5bb447dc230f488b8229ffa6d - [view on LGTM.com](https://lgtm.com/projects/g/MISP/MISP/rev/pr-883a843552d8eb99d2880087cd0a09858ea820ce)

**fixed alerts:**

* 1 for Missing variable declaration

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by mokaddem** [comment]
Recently fixed a bug that was occasionally preventing to show the picture if ``php-gd`` wasn't installed.
Could someone give it another try? @iglocska this time? :)

**Comment by iglocska** [comment]
Works for me with and without the extension, awesome job!

**Pull request merged**